### ISSUE TITLE
Core segfaults when rewind is enabled

### ISSUE THREAD

**Issue by ingolemo** [libretro/fuse-libretro#37]
When the _Settings > Rewind > Rewind Enable_ option is set to "On" in the menu, the core will segfault less than a second into playing any game. Here's a stack trace for the offending thread:

    #0  0x00007f7f1cb4104f raise (libc.so.6)
    #1  0x00007f7f1cb4247a abort (libc.so.6)
    #2  0x00007f7f1cb7ec50 __libc_message (libc.so.6)
    #3  0x00007f7f1cb84fe6 malloc_printerr (libc.so.6)
    #4  0x00007f7f1cb857de _int_free (libc.so.6)
    #5  0x00007f7f0de26a66 retro_serialize (fuse_libretro.so)
    #6  0x0000000000446bf6 n/a (retroarch)
    #7  0x0000000000428b3f n/a (retroarch)
    #8  0x0000000000419f2a n/a (retroarch)
    #9  0x00007f7f1cb2e291 __libc_start_main (libc.so.6)
    #10 0x00000000004172f9 n/a (retroarch)

The `retro_serialize` function appears to be freeing memory improperly. If you need a stack trace that includes the retroarch symbols, let me know.

I'm currently running this on Arch Linux, using the retroarch build from the arch's "community" repo (v1.3.6, x86_64) and the latest nightly fuse_libretro.so (as of 2016-10-18), although I can trigger this crash on several other version combinations.

**Comment by leiradel** [comment]
I've just pushed a commit that fixes snapshot memory management, but I'm not sure what to expect from the rewind functionality as I've never used it. Could you check if it works now? Thanks.

**Comment by ingolemo** [comment]
I don't really use the rewind functionality either, but I flipped that setting on (because why not) and was surprised sometime later when I was unable to use the fuse core. Once I tracked down the culprit I turned the setting off.

This commit does appear to fix the crash. The rewind functionality doesn't work correctly because when you press the rewind button you get sent to the very first frame that was displayed instead of rewinding (it's usually the title screen or some such, but if you auto-load a save-state on start-up it will use that instead).

Thank you.
